### ISSUE TITLE
pack_padded_sequence in rf.LSTM with default device='cuda' error

### ISSUE THREAD

**Issue by LucaG1** [rwth-i6/returnn#1406]
When using `rf.LSTM` with `rf.set_default_device_ctx("cuda")` I get an error with `pack_padded_sequence`.
``` 
result, (new_state_h, new_state_c) = source._raw_backend.lstm(                                                        
File "/u/luca.gaudino/setups/2023-08-10--rf-librispeech/returnn/returnn/torch/frontend/_backend.py", line 1734, in lstm                                                                                                                           source_packed = torch.nn.utils.rnn.pack_padded_sequence(source_raw, sizes_raw, enforce_sorted=False) 
File "/work/asr3/zeineldeen/hiwis/luca.gaudino/venv/env2/lib/python3.10/site-packages/torch/nn/utils/rnn.py", line 263, in pack_padded_sequence                                                                                                  
 _VF._pack_padded_sequence(input, lengths, batch_first)                                                              
RuntimeError: 'lengths' argument should be a 1D CPU int64 tensor, but got 1D cuda:0 Long tensor
```
I fixed it locally by moving sizes_raw to cpu here: https://github.com/rwth-i6/returnn/blob/6815e81eba4882fc249f8722e678cb7ad8ff3754/returnn/torch/frontend/_backend.py#L1729

So, I changed it to `sizes_raw = torch.reshape(sizes.raw_tensor, [batch_dim]).to('cpu')`.
But maybe it should be rather fixed in `pack_padded_sequence`.

**Comment by albertz** [comment]
Can you give some more context?

How are you using RF? What is your config file? Can you post the config and link it here? Can you also point out the relevant code?

Why are you using `set_default_device_ctx`? Normally that's already done by RETURNN and you don't need to care about it.

Why are the `lengths` on GPU? They always should be on CPU.

What is your `spatial_dim`?

**Comment by albertz** [comment]
Ah, I see. No, the problem should just be that we should have `device="cpu"` here for this case. I fixed this in e3b7babb8dcabdb7a31c5a5b7364797238067a1c. Can you try?

**Comment by LucaG1** [comment]
It works now, thank you!
